scripts/create-protection-market-convex.ts

```diff
@@ -0,0 +1,140 @@
+  if (!Object.keys(pools).includes(poolId)) throw new Error(`Pool ID ${poolId} is not a key in the \`pools\` object`);
+
+  const { contractName, name, symbol, description } = pools[poolId as keyof typeof pools];
+
+  // Verify recipient address was set properly
+  if (!utils.isAddress(recipient)) throw new Error('\n\n**** Please set the recipient address on line 23 ****\n');
+
+  // Compile contracts to make sure we're using the latest version of the trigger contracts
+  await hre.run('compile');
+
+  // Do some preparation
+  const underlyingAddress = getContractAddress('USDC', chainId);
+  const overrides = await getGasPrice();
+  const irModelAddress = '0x2B356b2ff9D6B001d51d6aF65A05946818F5e2E6'; // re-using IR model from Yearn Curve USDN market
+
+  // VERIFICATION
+  // Verify the user is ok with the provided inputs
+  console.log(chalk.bold.yellow('\nPLEASE VERIFY THE BELOW PARAMETERS\n'));
+  console.table({
+    'Deploying protection market for': `${contractName} Pool`,
+    'Deployer address:              ': `${signer.address}`,
+    'Deploying to network:          ': `${hre.network.name}`,
+    'Underlying token:              ': `${underlyingAddress}`,
+    'Gas price:                     ': `${JSON.stringify(overrides)}`,
+  });
+
+  const response = await waitForInput('\nDo you want to continue with deployment? y/N\n');
+  if (response !== 'y') {
+    logFailure('\nUser chose to cancel deployment. Exiting script');
+    return;
+  }
+  logSuccess('Continuing with deployment...\n');
+
+  // DEPLOY INTEREST RATE MODEL
+  // Using the one that's already deployed and defined as `irModelAddress` above
+
+  // DEPLOY TRIGGER
+  // Get instance of the Trigger ContractFactory with our signer attached
+  const triggerFactory: ContractFactory = await hre.ethers.getContractFactory(contractName, signer);
+
+  // Deploy the trigger contract (last constructor parameter is specific to the mock trigger contract)
+  const triggerParams = [name, symbol, description, platformIds, recipient, poolId];
+  const trigger: Contract = await triggerFactory.deploy(...triggerParams);
+  await trigger.deployed();
+  logSuccess(`${contractName} trigger deployed to ${trigger.address}`);
+
+  // VERIFY UNDERLYING
+  // We know that Money Markets have a trigger address of the zero address, so we use that to query the Comptroller
+  // for the Money Market address
+  const comptrollerAddress = getContractAddress('Comptroller', chainId);
+  const comptroller = new Contract(comptrollerAddress, comptrollerAbi, signer); // connect signer for sending transactions
+  const cozyMMAddress = await comptroller.getCToken(underlyingAddress, AddressZero);
+
+  // If the returned address is the zero address, a money market does not exist and we cannot deploy a protection
+  // market with the desired underlying
+  if (cozyMMAddress === AddressZero) {
+    logFailure('Money Market exists. Exiting script');
```

Comment:
this should be money market *doesn't* exist

Reply:
Oops, fixed in 69755fb

Reply:
fixed in 69755fb